frontend/src/components/table/useCheckboxTableBase.ts

```diff
@@ -7,6 +7,7 @@ type UseCheckboxTableBase<DataType> = {
   tableProps: Required<Pick<React.ComponentProps<typeof Table>, 'selectAll'>>;
   toggleSelection: (selection: DataType) => void;
   isSelected: (selection: DataType) => boolean;
+  disableCheck: (id: string, enabled: boolean) => void;
 };
 
 const useCheckboxTableBase = <T>(
```

Comment:
@DaoDaoNoCode made a mistake on the object reference side of things -- we need to make `data` pure.